nevergrad/common/decorators.py

```diff
@@ -3,28 +3,36 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
-from typing import Any, Callable, Optional, Dict
+from collections import UserDict
+from typing import Any, Callable, Optional, Dict, TypeVar
 import functools
 
 
-class Registry(dict):  # type: ignore
+X = TypeVar("X")
+
+
+# pylint does not understand Dict, and mypy needs Dict to be generic...
+# the following is (very) hacky but solves both issues for now
+# It can be safely reverted to Registry(dict) though...
+# pylint: disable=too-many-ancestors
+class Registry(Dict[str, X], UserDict):  # type: ignore
```

Comment:
I don't think this does what you want. Dict means this is a *real* dict, whereas UserDict means that this has a dict as the `data` attribute. They define exactly the same methods, so the one that comes in the MRO overrides the second one. What this means here is that UserDict has no effect at all. If you want to keep UserDict make the first a MutableMapping.

Reply:
It actually *does* what I want but it is really hacky:
- the methods which are used are the ones in `Dict[str, X]`, which is convenient since they are typed.
- `UserDict` is basically only there so that pylint does not complain since it does not recognize Dict. The `data` field is unused... actually it does not even exist, not sure why :s
- using `Registry(UserDict, MutableMapping[str, X])` has untyped getitem... I would have to rewrite all dict methods :s

Reply:
You can have this extend just MutableMapping, remove all dict or UserDict superclass, then manually define a `data` attribute, which is a dict, and manually implement the 5 abstract methods of MutableMapping (see there [here](https://docs.python.org/3/library/collections.abc.html)). These methods will basically just be
```py
def foo(*args, *kwargs):
    return self.data.foo(*args, **kwargs)
```
Once you do so you get for free all other methods of the dictionary interface. So, in short, your class will be a regular class (not a dict) but will be dict-like.

Reply:
I was a bit afraid pylint would not recognize it either, but it worked perfectly! thank you!